File: templates/lang/kotlin_android_app_dagger/src_files/app/AppComponent.java
package {{ .Group }}.app;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
    AndroidSupportInjectionModule.class,
    {{ .CamelNameWithoutApp }}AppModule.class,
    RootBindingModule.class})
interface {{ .CamelNameWithoutApp }}AppComponent extends AndroidInjector<{{ .CamelNameWithoutApp }}App> {

  @Component.Builder
  abstract class Builder extends AndroidInjector.Builder<{{ .CamelNameWithoutApp }}App> {}
}

File: templates/lang/kotlin_android_app_dagger/src_files/app/AppComponent.kt
package {{ .Group }}.app

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
    AndroidSupportInjectionModule::class,
    {{ .CamelNameWithoutApp }}AppModule::class,
    RootBindingModule::class))
interface {{ .CamelNameWithoutApp }}AppComponent : AndroidInjector<{{ .CamelNameWithoutApp }}App> {

  @Component.Builder
  abstract class Builder : AndroidInjector.Builder<{{ .CamelNameWithoutApp }}App>()
}
